### ISSUE TITLE
[Bug] Fail to read the raw string of the assets

### ISSUE THREAD

**Issue by niuiic** [farm-fe/farm#652]
### Steps to reproduce

- `main.ts`

```typescript
import fsSource from './test.frag?raw'
import vsSource from './test.vert?raw'
...(not important)
```

- `test.frag`

```glsl
void main() {
  gl_FragColor = vec4(1.0, 0.0, 1.0, 1.0);
}
```

- `test.vert`

```glsl
attribute vec4 aVertexPosition;

void main(void ) {
  gl_Position = aVertexPosition;
}
```

### What is expected?

import file content as string

### What is actually happening?

```
webgl:build: Parse `src/test.frag?raw` failed.
webgl:build:  Error: 
webgl:build:   x Unterminated string constant
webgl:build:    ,-[src/test.frag?raw:1:1]
webgl:build:  1 | export default "void main() {
webgl:build:    :                ^^^^^^^^^^^^^^
webgl:build:  2 |   gl_FragColor = vec4(1.0, 0.0, 1.0, 1.0);
webgl:build:  3 | }
webgl:build:  4 | "
webgl:build:    `----
webgl:build: 
webgl:build:   x Expression expected
webgl:build:    ,-[src/test.frag?raw:1:1]
webgl:build:  1 | export default "void main() {
webgl:build:  2 |   gl_FragColor = vec4(1.0, 0.0, 1.0, 1.0);
webgl:build:  3 | }
webgl:build:    : ^
webgl:build:  4 | "
webgl:build:    `----
```

### System Info

```Shell
System:
    OS: Linux 6.1 Gentoo Linux
    CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
    Memory: 11.23 GB / 15.59 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.6.1 - /run/user/1000/fnm_multishells/242734_1698043783642/bin/node
    npm: 9.8.1 - /run/user/1000/fnm_multishells/242734_1698043783642/bin/npm
    pnpm: 8.9.2 - /run/user/1000/fnm_multishells/242734_1698043783642/bin/pnpm
  npmPackages:
    @farmfe/core: ^0.13.10 => 0.13.10
```

### Any additional comments?

A simple js plugin which works. But I cannot compile it with farm correctly.

- plugin

```typescript
import type { DevServer, JsPlugin, UserConfig } from '@farmfe/core'
import { promises } from 'fs'

export const farmRawPlugin = (fileSuffixes: string[]): JsPlugin => {
  let farmConfig: UserConfig['compilation']
  let devServer: DevServer

  return {
    name: 'raw-resolve-plugin',
    config: (args) => (farmConfig = args),
    configDevServer: (args) => (devServer = args),
    load: {
      filters: {
        resolvedPaths: fileSuffixes.map((x) => `\\.${x}$`)
      },
      executor: async (args) => {
        const content = await promises.readFile(args.resolvedPath, 'utf8')
        return {
          content,
          moduleType: 'raw'
        }
      }
    },
    transform: {
      filters: {
        moduleTypes: ['raw']
      },
      executor: (args) => {
        if (farmConfig?.mode === 'development') {
          devServer?.addWatchFile(args.resolvedPath, [])
        }
        return { content: 'export default' + '`' + args.content + '`', moduleType: 'js' }
      }
    }
  }
}
```

- farm.config.ts for this plugin

```typescript
import type { UserConfig } from '@farmfe/core'
import farmDtsPlugin from '@farmfe/js-plugin-dts'
import { builtinModules } from 'node:module'

const config: UserConfig = {
  compilation: {
    input: {
      index: 'src/index.ts'
    },
    output: {
      path: 'dist',
      entryFilename: '[entryName].cjs',
      targetEnv: 'node',
      format: 'cjs'
    },
    external: [...builtinModules.map((x) => `^${x}$`), ...builtinModules.map((x) => `^node:${x}$`)],
    partialBundling: {
      enforceResources: [
        {
          name: 'index.js',
          test: ['.+']
        }
      ]
    },
    minify: false,
    sourcemap: false,
    presetEnv: false
  },
  plugins: [
    farmDtsPlugin({
      tsConfigPath: './tsconfig.json'
    })
  ]
}

export default config
```

- package.json for this plugin

```json
{
  "name": "farm-plugins",
  "version": "0.0.0",
  "main": "dist/index.cjs",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "farm build"
  },
  "devDependencies": {
    "tsconfig": "workspace:*",
    "eslint-config-custom": "workspace:*"
  }
}
```

- use it in another package

```typescript
import type { UserConfig } from '@farmfe/core'
import { farmRawPlugin } from 'farm-plugins'
import path from 'path'

const config: UserConfig = {
  compilation: {
    resolve: {
      alias: {
        '@/': path.join(process.cwd(), 'src')
      }
    }
  },
  plugins: [farmRawPlugin(['vert', 'frag'])]
}

export default config
```

Build error.

```
webgl:build: cache bypass, force executing 4e8b574b937c8f15
webgl:build: 
webgl:build: > webgl@0.0.0 build /home/niuiic/Documents/projects/gallery/apps/webgl
webgl:build: > farm build
webgl:build: 
webgl:build: [ Farm ] Using config file at /home/niuiic/Documents/projects/gallery/apps/webgl/farm.config.ts
webgl:build: file:///home/niuiic/Documents/projects/gallery/apps/webgl/node_modules/.farm/farm.config.bundle1698044539690.mjs:488
webgl:build:                 (0, _farmplugins.farmRawPlugin)([
webgl:build:                                                ^
webgl:build: 
webgl:build: TypeError: (0 , _farmplugins.farmRawPlugin) is not a function
webgl:build:     at farm.config.ts (file:///home/niuiic/Documents/projects/gallery/apps/webgl/node_modules/.farm/farm.config.bundle1698044539690.mjs:488:48)
webgl:build:     at ModuleSystem.require (file:///home/niuiic/Documents/projects/gallery/apps/webgl/node_modules/.farm/farm.config.bundle1698044539690.mjs:96:17)
webgl:build:     at file:///home/niuiic/Documents/projects/gallery/apps/webgl/node_modules/.farm/farm.config.bundle1698044539690.mjs:497:166
webgl:build:     at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
webgl:build:     at async ModuleLoader.import (node:internal/modules/esm/loader:308:24)
webgl:build:     at async importFresh (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.13.10/node_modules/@farmfe/core/dist/utils/share.js:56:19)
webgl:build:     at async readConfigFile (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.13.10/node_modules/@farmfe/core/dist/config/index.js:311:20)
webgl:build:     at async resolveUserConfig (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.13.10/node_modules/@farmfe/core/dist/config/index.js:233:28)
webgl:build:     at async build (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.13.10/node_modules/@farmfe/core/dist/index.js:50:24)
webgl:build: 
webgl:build: Node.js v20.6.1
webgl:build:  ELIFECYCLE  Command failed with exit code 1.
webgl:build: ERROR: command finished with error: command (/home/niuiic/Documents/projects/gallery/apps/webgl) pnpm run build exited (1)
```

**Comment by ErKeLost** [comment]
Can you provide a demo? to ensure some uncertainties, we may not be able to reproduce your problem.

**Comment by wre232114** [comment]
For the first issue, that's a bug, we will fix it now.

For the second issue, you should export your plugin using `default`, from example:
```ts
export default const farmRawPlugin = (fileSuffixes: string[]): JsPlugin => {
// ...
}
```

and import them as `default` too:
```ts
import farmRawPlugin from 'farm-plugins'
```

**Comment by wre232114** [comment]
BTW, you does not need to transform you module into `asset` first. you can just load `test.frag` in your plugin, and everything will work too.

**Comment by niuiic** [comment]
Thank you for help, now I can build my plugin correctly.

**Comment by wre232114** [comment]
issue 1 is fixed and issue 2 is supported by #655
